clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1105,6 +1105,18 @@ static ParmVarDecl *getSyclKernelHandlerArg(FunctionDecl *KernelCallerFunc) {
   return (KHArg != KernelCallerFunc->param_end()) ? *KHArg : nullptr;
 }
 
+static bool isReadOnlyAccessor(const TemplateArgument AccessModeArg) {
+  const auto *AccessModeArgEnumType =
+      AccessModeArg.getIntegralType()->castAs<EnumType>();
+
+  for (auto *E : AccessModeArgEnumType->getDecl()->enumerators())
+    if (E->getName() == "read" &&
```

Comment:
Enum namescan't be duplicated, so you can make this all:
```suggestion
    if (E->getName() == "read") return E->getInitVal() == AccessModeArg.getAsIntegral();
```